okhttp/src/main/java/okhttp3/internal/http1/Http1Codec.java

```diff
@@ -196,6 +196,9 @@ public void writeRequest(Headers headers, String requestLine) throws IOException
 
       if (expectContinue && statusLine.code == HTTP_CONTINUE) {
         return null;
+      } else if (statusLine.code == HTTP_CONTINUE) {
```

Comment:
I don't understand the need for this branch since it duplicates the lines below

Reply:
It sets the state to a different one (STATE_OPEN_REQUEST_BODY instead of STATE_OPEN_RESPONSE_BODY). Otherwise, sending the response body again would not work.

Reply:
It's early, sorry!